src/main/java/org/verdictdb/coordinator/ExecutionContext.java

```diff
@@ -94,33 +94,60 @@ public VerdictResultStream streamsql(String query) throws VerdictDBException {
       updateDefaultSchemaFromQuery(query);
       return null;
     } else if (queryType.equals(QueryType.show_databases)) {
-      return generateShowSchemaResultFromQuery();
+      VerdictResultStream stream = generateShowSchemaResultFromQuery();
+      return stream;
     } else if (queryType.equals(QueryType.show_tables)) {
-      return generateShowTablesResultFromQuery(query);
+      VerdictResultStream stream = generateShowTablesResultFromQuery(query);
+      return stream;
     } else if (queryType.equals(QueryType.describe_table)) {
-      return generateDesribeTableResultFromQuery(query);
+      VerdictResultStream stream = generateDesribeTableResultFromQuery(query);
+      return stream;
     } else {
       throw new VerdictDBTypeException("Unexpected type of query: " + query);
     }
   }
 
 
   private VerdictResultStream generateShowSchemaResultFromQuery() throws VerdictDBException {
-    VerdictSingleResultFromListData result = new VerdictSingleResultFromListData((List)context.getConnection().getSchemas());
+    VerdictSingleResultFromListData result = new VerdictSingleResultFromListData(QueryType.show_databases, (List) context.getConnection().getSchemas());
```

Comment:
Let's just pass a list here. For example,
```java
List<String> header = Collections.singletonList("schema");
List<List<String>> rows = Collections.singletonList(
    context.getConnection().getSchemas());
VerdictSingleResultFromListData result = 
    new VerdictSingleResultFromListData(header, rows);
```

Please update `VerdictSingleResultFromListData`'s constructor accordingly.